enf-reducer/tests/cut-unnecessary-or-test.metta

```diff
@@ -0,0 +1,44 @@
+! (register-module! ../../../metta-moses-reduction)
+! (import! &self metta-moses-reduction:enf-reducer:cut-unnecessary-or)
+
+ ;; Test 1: OR with multiple top-level children, sub-expressions not handled (no transformation)
+! (assertEqual (orCut (OR a (OR b) (OR c (OR f)))) (OR a (OR b) (OR c (OR f))))
```

Comment:
This test case is invalid. We won't have a consecutive `OR`s as input for this function. This is due to the `gatherJunctors` algorithm which will make sure that any boolean expression has an alternating sequence of `OR`s and `AND`s always. There fore, The above test case is as simple as: `(OR a b c f)`. Make sure to update your test cases as such.

Reply:
Also make sure to add the test case found on the holman paper. And other similar cases where things get complicated. i.e., `(OR (AND (NOT A) B C D)` and `(OR (AND (NOT A) B C D (OR A B))`.